src/services/util/builder.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Get root path from an optional String.
+pub fn root_path(root: &Option<String>) -> String {
```

Comment:
I prefer to keep those path related functions as is (at least in this PR).

Reply:
copy the logic from azblob

Reply:
> copy the logic from azblob

That's OK. Please keep inside `obs` mod instead of `util`.